packages/nvidia_nat_rag_lib/src/nat/plugins/rag_lib/client.py

```
@@ -0,0 +1,216 @@
+    rag_config.llm = rag_config.llm.model_copy(update=base_dict)
+    rag_config.reflection = rag_config.reflection.model_copy(update=base_dict)
+    rag_config.filter_expression_generator = rag_config.filter_expression_generator.model_copy(update=base_dict)
+
+    # resolve embedder config
+    nim_embedder_config = builder.get_embedder_config(config.embedder)
+    if not isinstance(nim_embedder_config, NIMEmbedderModelConfig):
+        raise ValueError(f"Unsupported embedder config type: {type(config.embedder)}. Expected NIMEmbedderModelConfig.")
+    base_dict = nim_embedder_config.model_dump(include={"base_url", "model_name", "api_key", "dimensions"},
+                                               exclude_none=True)
+    if "base_url" not in base_dict:
+        raise ValueError("base_url is required for embedder config specified in in NVIDIA RAG Config.")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)